Mesh_3/include/CGAL/Mesh_3/Detect_features_in_image.h

```diff
@@ -0,0 +1,290 @@
+            std::cerr << "vu: " << vu << "\n";
+            std::cerr << "vv: " << vv << "\n";
+            std::cerr << "vw: " << vw << "\n";
+#endif // CGAL_DEBUG_TRIPLE_LINES
+            for (auto& polyline : cube_features) {
+              for (auto& point : polyline) {
+                point = pa
+                  + point.x() * vu
+                  + point.y() * vv
+                  + point.z() * vw;
+                point = { vx * point.x(),
+                          vy * point.y(),
+                          vz * point.z(), };
+                point = point + translation;
+              }
+              for (int i = 0; i < 2; ++i) {
+                Point_3& extremity = (i == 0) ? polyline.front() : polyline.back();
+                Vertex_handle vh = triangulation.nearest_vertex(extremity, start_cell);
+                if (Vertex_handle() != vh) {
+                  if (squared_distance(vh->point(), extremity) < sq_dist_bound) {
+                    extremity = vh->point();
+                  }
+                }
+                vh = triangulation.insert(extremity, start_cell);
+                start_cell = vh->cell();
+              }
+              features_inside.push_back(std::move(polyline));
+            } // end loop on polylines
+          } // end case where the transformation is not the identity
+        } // end if the reference_cube has polylines
+      }
+
+  // call the split_graph_into_polylines, to create long polylines from the
+  // short polylines that were generated per voxel.
+  Polylines polylines_inside;
+  CGAL::polylines_to_protect(polylines_inside,
+                             features_inside.begin(),
+                             features_inside.end());
+
+  Polylines polylines_on_bbox;
+  CGAL::polylines_to_protect<Point_3, Word_type>(image, polylines_on_bbox,
+                                                 polylines_inside.begin(),
+                                                 polylines_inside.end());
+
+  polylines_inside.insert(polylines_inside.end(),
+                          polylines_on_bbox.begin(),
+                          polylines_on_bbox.end());
+
+#ifdef CGAL_DEBUG_TRIPLE_LINES
+  std::ofstream output_polylines("out-generated.polylines.txt");
+  output_polylines.precision(17);
+  for (auto poly : boost::range::join(polylines_on_bbox, polylines_inside)) {
+    output_polylines << poly.size();
+    for (auto p : poly) output_polylines << " " << p;
+    output_polylines << std::endl;
+  }
+#endif
+
+  return polylines_inside;
+}
+
+
+}// namespace internal
+
+/*!
+* \ingroup PkgMesh3FeatureDetection
+*
+* Functor for feature detection in labeled images.
+*/
+struct Detect_features_in_image
+{
+public:
+  /*!
+  * detects and constructs the polylines that lie at the
+  * intersection of three or more subdomains.
+  *
+  * Each subdomain inside the bounding box
+  * of the input labeled image is defined as the set of voxels
+  * with the same value. The outside of the bounding box
```

Comment:
That is not completely true. Is the trilinear interpolation explained somewhere?

Reply:
I don't think so...